**Task 1 : Data Overview**

Step 1 : Business problem undertanding

Objective: Understand the dataset structure.


In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv("Data_set.csv")

In [3]:
# display top 5 rows
df.head()

gender  age Investment_Avenues  Mutual_Funds  Equity_Market  Debentures  \
0  Female   34                Yes             1              2           5   
1  Female   23                Yes             4              3           2   
2    Male   30                Yes             3              6           4   
3    Male   22                Yes             2              1           3   
4  Female   24                 No             2              1           3   

   Government_Bonds  Fixed_Deposits  PPF  Gold  ...           Duration  \
0                 3               7    6     4  ...          1-3 years   
1                 1               5    6     7  ...  More than 5 years   
2                 2               5    1     7  ...          3-5 years   
3                 7               6    4     5  ...   Less than 1 year   
4                 6               4    5     7  ...   Less than 1 year   

  Invest_Monitor   Expect       Avenue What are your savings objectives?  \
0        Monthly  20%-30%  Mutual Fund                   Retirement Plan   
1         Weekly  20%-30%  Mutual Fund                       Health Care   
2          Daily  20%-30%       Equity                   Retirement Plan   
3          Daily  10%-20%       Equity                   Retirement Plan   
4          Daily  20%-30%       Equity                   Retirement Plan   

          Reason_Equity         Reason_Mutual     Reason_Bonds  \
0  Capital Appreciation        Better Returns  Safe Investment   
1              Dividend        Better Returns  Safe Investment   
2  Capital Appreciation          Tax Benefits  Assured Returns   
3              Dividend  Fund Diversification   Tax Incentives   
4  Capital Appreciation        Better Returns  Safe Investment   

             Reason_FD                    Source  
0        Fixed Returns  Newspapers and Magazines  
1  High Interest Rates     Financial Consultants  
2        Fixed Returns                Television  
3  High Interest Rates                  Internet  
4            Risk Free                  Internet  

[5 rows x 24 columns]

In [4]:
# display last 5 rows
df.tail()

gender  age Investment_Avenues  Mutual_Funds  Equity_Market  Debentures  \
35   Male   30                Yes             1              4           6   
36   Male   30                Yes             2              4           7   
37   Male   25                Yes             5              4           7   
38   Male   31                Yes             2              4           7   
39   Male   29                Yes             4              3           5   

    Government_Bonds  Fixed_Deposits  PPF  Gold  ...   Duration  \
35                 5               3    2     7  ...  3-5 years   
36                 5               1    3     6  ...  1-3 years   
37                 6               1    2     3  ...  3-5 years   
38                 5               3    1     6  ...  1-3 years   
39                 7               2    1     6  ...  3-5 years   

   Invest_Monitor   Expect                 Avenue  \
35        Monthly  20%-30%         Fixed Deposits   
36        Monthly  20%-30%                 Equity   
37        Monthly  30%-40%  Public Provident Fund   
38         Weekly  20%-30%                 Equity   
39        Monthly  20%-30%         Fixed Deposits   

   What are your savings objectives?         Reason_Equity  \
35                       Health Care  Capital Appreciation   
36                   Retirement Plan  Capital Appreciation   
37                       Health Care  Capital Appreciation   
38                       Health Care              Dividend   
39                   Retirement Plan              Dividend   

           Reason_Mutual     Reason_Bonds      Reason_FD  \
35        Better Returns  Assured Returns  Fixed Returns   
36        Better Returns  Assured Returns      Risk Free   
37        Better Returns  Safe Investment  Fixed Returns   
38  Fund Diversification  Assured Returns  Fixed Returns   
39        Better Returns  Safe Investment  Fixed Returns   

                      Source  
35     Financial Consultants  
36  Newspapers and Magazines  
37     Financial Consultants  
38  Newspapers and Magazines  
39     Financial Consultants  

[5 rows x 24 columns]

In [5]:
# Gives information about dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 24 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   gender                             40 non-null     object
 1   age                                40 non-null     int64 
 2   Investment_Avenues                 40 non-null     object
 3   Mutual_Funds                       40 non-null     int64 
 4   Equity_Market                      40 non-null     int64 
 5   Debentures                         40 non-null     int64 
 6   Government_Bonds                   40 non-null     int64 
 7   Fixed_Deposits                     40 non-null     int64 
 8   PPF                                40 non-null     int64 
 9   Gold                               40 non-null     int64 
 10  Stock_Marktet                      40 non-null     object
 11  Factor                             40 non-null     object
 12  Objective 

In [6]:
# Display all columns of dataset
df.columns

Index(['gender', 'age', 'Investment_Avenues', 'Mutual_Funds', 'Equity_Market',
       'Debentures', 'Government_Bonds', 'Fixed_Deposits', 'PPF', 'Gold',
       'Stock_Marktet', 'Factor', 'Objective', 'Purpose', 'Duration',
       'Invest_Monitor', 'Expect', 'Avenue',
       'What are your savings objectives?', 'Reason_Equity', 'Reason_Mutual',
       'Reason_Bonds', 'Reason_FD', 'Source'],
      dtype='object')

In [7]:
# Display all columns of dataset
df.describe()

age  Mutual_Funds  Equity_Market  Debentures  Government_Bonds  \
count  40.000000     40.000000      40.000000   40.000000         40.000000   
mean   27.800000      2.550000       3.475000    5.750000          4.650000   
std     3.560467      1.197219       1.131994    1.675617          1.369072   
min    21.000000      1.000000       1.000000    1.000000          1.000000   
25%    25.750000      2.000000       3.000000    5.000000          4.000000   
50%    27.000000      2.000000       4.000000    6.500000          5.000000   
75%    30.000000      3.000000       4.000000    7.000000          5.000000   
max    35.000000      7.000000       6.000000    7.000000          7.000000   

       Fixed_Deposits        PPF       Gold  
count       40.000000  40.000000  40.000000  
mean         3.575000   2.025000   5.975000  
std          1.795828   1.609069   1.143263  
min          1.000000   1.000000   2.000000  
25%          2.750000   1.000000   6.000000  
50%          3.500000   1.000000   6.000000  
75%          5.000000   2.250000   7.000000  
max          7.000000   6.000000   7.000000

**Dataset Overview :**

* Number of Entries : 40 entries

* Number of Columns : 24 columns



* Data set :
* Categorical (Enum) Columns:

gender: ['Male', 'Female']

Investment_Avenues: ['Yes', 'No']

Stock_Marktet: ['Yes', 'No']

Factor: ['Returns', 'Risk', 'Locking Period']

Objective: ['Growth', 'Income']

Purpose: ['Wealth Creation', 'Savings for Future', 'Returns']

Duration: ['3-5 years', '1-3 years', 'Less than 1 year', 'More than 5 years']

Invest_Monitor: ['Monthly', 'Weekly', 'Daily']

Expect: ['20%-30%', '30%-40%', '10%-20%']

Avenue: ['Mutual Fund', 'Equity', 'Fixed Deposits']

What are your savings objectives?: ['Retirement Plan', 'Health Care', 'Education']

Reason_Equity: ['Dividend', 'Liquidity']

Reason_Mutual: ['Better Returns', 'Tax Benefits']

Reason_Bonds: ['Assured Returns', 'Safe Investment', 'Tax Incentives']

Reason_FD: ['Risk Free', 'Fixed Returns', 'High Interest Rates']

Source: ['Television', 'Internet']


* Numerical (Int64) Columns:

age: Mean = 27.80, Std = 3.56, Min = 21.00, Max = 35.00

Mutual_Funds: Mean = 2.55, Std = 1.20, Min = 1.00, Max = 7.00

Equity_Market: Mean = 3.48, Std = 1.13, Min = 1.00, Max = 6.00

Debentures: Mean = 5.75, Std = 1.68, Min = 1.00, Max = 7.00

Government_Bonds: Mean = 4.65, Std = 1.37, Min = 1.00, Max = 7.00

Fixed_Deposits: Mean = 3.58, Std = 1.80, Min = 1.00, Max = 7.00

PPF: Mean = 2.02, Std = 1.61, Min = 1.00, Max = 6.00

Gold: Mean = 5.97, Std = 1.14, Min = 2.00, Max = 7.00

**Insights :**
* Counts: There are 25 males and 15 females in the dataset.
* Mean and Standard Deviation of gender : The mean count is 20.00 with a standard deviation of 7.07.
* Age Range: The ages in the dataset range from 21 to 35 years.
* Mean and Standard Deviation pf age : The mean age is 27.80 with a standard deviation of 3.56.
* Detailed Records: The dataset includes 40 age records with varying frequencies.
---
* Invesments Counts: There are 37 'Yes' and 3 'No' responses.
* Investment Avenues: The dataset includes seven investment avenues: 'Mutual_Funds', 'Equity_Market', 'Debentures', 'Government_Bonds', 'Fixed_Deposits', 'PPF', and 'Gold'.
